pkgs/build-support/setup-hooks/patch-desktop-file-exec.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# the shebang is here for shellcheck
+
+# This setup hook will do the following to any installed desktop file:
+# replace Exec=/any/path/to/foo by a shell script lauching
+# * $out/bin/foo if it exists
+# * foo (from $PATH) otherwise
+# This transformation is only done if
+# $out/bin/foo exists and is executable
+# It can also pick foo in buildInputs.
+# The goal is that:
+# * no manual patching is needed for desktop file which hardcode /usr/bin/foo
+# * if the store path containing foo still exists, then it is used
+# * if the desktop file is copied (usually, done by the menu editor of desktop environments),
+# foo is updated and then the original store path containing foo is garbage-collected,
+# the desktop file still executes foo from $PATH.
+
+fixupOutputHooks+=('autoPatchDesktopFileExec;')
+
+autoPatchDesktopFileExec () {
+    # shellcheck disable=2154
+    dir="${!outputBin}/share/applications";
+    if [[ -d $dir ]]; then
+        patchDesktopFileExecIn "$dir";
+    fi;
+}
+
+patchDesktopFileExecIn () {
+    echo "fixing Exec= paths in desktop files of $1";
```

Comment:
```suggestion
    echo "fixing Exec= paths in desktop files of $1"
```
bash does not require end of line ; except when having multiple commands on one line.